### DISCUSSION TITLE
Create SQL Query based Scheduling for Data Sync

### DISCUSSION THREAD

**Discussion by shrivatsahosabettu** [dbeaver/dbeaver#22834] [Q&A]
### Description

Is there any feature or facility to sync the data based on the query. Currently we can copy the whole Table from one schema to another one. 
The reason for asking is every day if we want to copy the data on additional data on the first schema to the second one. It will be redundant data picking with the task scheduler on database copy activity.

### DBeaver Version

Community Edition Version 23.3.4

### Operating System

Windows 10

### Database and driver

DB2 Z/OS

### Steps to reproduce

NA

### Additional context

NA

**Comment by ShadelessFox** [comment]
Hello @shrivatsahosabettu,

DBeaver has a data transfer and task scheduling feature. This might be something you're interested in.

Please refer to the wiki for more information:
- https://github.com/dbeaver/dbeaver/wiki/Data-transfer
- https://github.com/dbeaver/dbeaver/wiki/Data-transfer
- https://github.com/dbeaver/dbeaver/wiki/Task-Scheduler

**Comment by shrivatsahosabettu** [comment]
@ShadelessFox  This is what I know as data transfer and Task Scheduler. What I want to know can we Schedule the task with SQL which we already use for CSV export and import?

Example: 

`SELECT * FROM EMPLOYEE
WHERE EMP_JOIN_DATE = CURRENT DATE`

Can I create a task on this SQL statement to run and everyday I get data based on the date incremented I get the recent loaded data in the EMPLOYEE table from one Schema to another schema.

**Comment by ShadelessFox** [comment]
If I understand you correctly, you can create a composite task consisting of an export task that will produce a file based on your query, an an import task that will import that file into your database.

**Comment by shrivatsahosabettu** [comment]
Let me make it more clear. When we do the data transfer probably a back end a bulk load is happening on the whole table data load from one schema to another. The same thing can we achieve using a SQL based instead of whole table loading from one schema to another?

**Comment by ShadelessFox** [comment]
It's up to the database in the question. For DB2, something like this should work:
```sql
CREATE TABLE NEW_TABLE LIKE OLD_TABLE;
INSERT INTO NEW_TABLE (SELECT * FROM OLD_TABLE);
```

**Comment by shrivatsahosabettu** [comment]
Yes certainly we can do like that. But when I don't have a CREATE Table access, and I have to load data from PROD to Development region DB's Table where I need to clone certain set of data is there an option to schedule with a SQL? Currently I do follow with CSV export(from Prod) and Import(to Dev) on the new data which is posted everyday in Production copy to Development to test it.

If I have SQL based scheduling method I just thinking is to write a SQL where row_created_timestamp/ Row_Updated_Timestamp = current date on the table and everyday new set of data will be copied automatically to my Development region on the time.

**Comment by Destrolaric** [comment]
@shrivatsahosabettu We are unaware of the ability to do something similar with SQL in the DB2. Unfortunately, I am not sure we can help.
